Source/SwiftLintFramework/Rules/Lint/IdenticalOperandsRule.swift

```diff
@@ -48,57 +50,148 @@ public struct IdenticalOperandsRule: ConfigurationProviderRule, OptInRule, Autom
-                let range = result.range(at: 1)
-                return !range.intersects(exclusionRanges)
-            }
-            .filter { result, tokens in
-                let contents = file.contents.bridge()
-                let range = result.range(at: 1)
-                guard let byteRange = contents.NSRangeToByteRange(start: range.location,
-                                                                  length: range.length) else {
-                    return false
+        return
+            file.matchesAndTokens(matching: "\\s(" + operators + ")\\s")
+                .filter { _, tokens in tokens.isEmpty }
+                .compactMap { matchResult, _ in violationRangeFrom(match: matchResult, in: file) }
+                .map { range in
+                    return StyleViolation(ruleDescription: type(of: self).description,
+                                          severity: configuration.severity,
+                                          location: Location(file: file, characterOffset: range.location))
                 }
+    }
+
+    private func violationRangeFrom(match: NSTextCheckingResult, in file: File) -> NSRange? {
+        let contents = file.contents.bridge()
+        let operatorRange = match.range(at: 1)
+        guard let operatorByteRange = contents.NSRangeToByteRange(operatorRange) else {
+            return nil
+        }
+
+        let tokens = file.syntaxMap.tokens
+        guard let rightTokenIndex = tokens.firstIndex(where: { $0.offset >= operatorByteRange.upperBound }),
+            rightTokenIndex > 0 else {
+                return nil
+        }
+
+        let (leftOperand, rightOperand) = operandsStartingFromIndexes(leftTokenIndex: rightTokenIndex - 1,
+                                                                      rightTokenIndex: rightTokenIndex,
+                                                                      file: file)
+
+        guard leftOperand.tokens.count == rightOperand.tokens.count else {
+            return nil
+        }
 
-                let kinds = tokens
-                    .filter { $0.offset >= byteRange.location }
-                    .kinds
-                return syntaxKinds.isDisjoint(with: kinds)
+        // Make sure that there's nothing but operator between tokens
+        let operatorString = contents.substring(with: operatorRange)
+        guard let leftToken = leftOperand.tokens.last, let rightToken = rightOperand.tokens.first else {
+            return nil
+        }
+        guard contents.isRegexBetweenTokens(leftToken, operatorString, rightToken) else {
+            return nil
+        }
+
+        // Make sure both operands have same token types
+        guard zip(leftOperand.tokens, rightOperand.tokens).allSatisfy({ $0.0.type == $0.1.type }) else {
+            return nil
+        }
+
+        // Make sure that every part of the operand part is equal to previous on
+        guard zip(leftOperand.tokens, rightOperand.tokens).allSatisfy({
```

Comment:
Why not use `==` instead of zip+allSatisfy?

```swift
guard leftOperand.tokens.map(contents.subStringWithSyntaxToken) ==
    rightOperand.tokens.map(contents.subStringWithSyntaxToken) else {
    return nil
}
```

Reply:
¯\\_(ツ)\_/¯